src/core/nis1/Nis1Transaction.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Deadline, Network, Nis1KeyPair, Nis1TransactionType } from '@core';
+import { Duration, Instant } from '@js-joda/core';
+import { toBufferLE } from 'bigint-buffer';
+import { Key } from '../Key';
+import { Converter } from '../utils/Converter';
+import { Serializer } from './transactions/Serializer';
+
+/**
+ * Generic Nis1 transaction class
+ */
+export class Nis1Transaction<T extends Serializer> {
```

Comment:
Has nis1 transaciton a transaction hash?

Reply:
don't think it has generation hash, but transaction hash yes, but this PR is just following python sdk for now, (until we have catbuffer schema ready)

Reply:
nope, nis1 doesn't have a generation hash.